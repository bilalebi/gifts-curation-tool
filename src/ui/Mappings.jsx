import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import ResultsTable from './components/ResultsTable';

import '../styles/Home.css';

class Mappings extends Component {
  constructor(props) {
    super(props);
    this.state = {
      offset: 0,
      limit: 15,
      activeFacets: {},
      initialPage: 0,
    };
  }

  getFacetsAsString = () =>
    Object.keys(this.state.activeFacets)
      .map(key => `${key}:${this.state.activeFacets[key]}`)
      .join(',');

  removeFilter = (facet) => {
    const { activeFacets } = this.state;
    delete activeFacets[facet];
    this.setState({
      activeFacets,
    });
  };

  addFilter = (facet, value) => {
    const { activeFacets } = this.state;
    activeFacets[facet] = value;
    this.setState({
      activeFacets,
    });
  };

  handlePageClick = data => {
    const initialPage = data.selected;
    const offset = Math.ceil(initialPage * this.state.limit);
    this.setState({
      offset,
      initialPage
    });
  };

  render() {
    const propsToPass = {
      addFilter: this.addFilter,
      removeFilter: this.removeFilter,
      handlePageClick: this.handlePageClick,
      activeFacets: this.state.activeFacets,
      params: {
        searchTerm: this.props.searchTerm,
        facets: this.getFacetsAsString(this.state.activeFacets),
        offset: this.state.offset,
        limit: this.state.limit,
        format: 'json',
      },
      history: this.props.history,
      initialPage: this.state.initialPage,
      clearSearchTerm: this.props.clearSearchTerm,
    };
    return <ResultsTable {...propsToPass} />;
  }
}

Mappings.propTypes = {
  searchTerm: PropTypes.string,
};

Mappings.defaultProps = {
  searchTerm: '',
};

export default withRouter(Mappings);
