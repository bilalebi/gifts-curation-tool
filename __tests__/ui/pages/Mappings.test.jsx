import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Mappings from '../../../src/ui/Mappings';

describe('Mappings component', () => {
  test('should render', () => {
    const component = renderer
      .create(<MemoryRouter><Mappings /></MemoryRouter>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
