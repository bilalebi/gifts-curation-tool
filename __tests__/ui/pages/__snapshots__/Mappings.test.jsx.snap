// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mappings component should render 1`] = `
<div
  className="loading-spinner"
>
  <div
    className="css-1d8ny0s"
  >
    <div
      className="css-1hs6dmj"
    />
    <div
      className="css-1apkdvz"
    />
    <div
      className="css-ruz37k"
    />
    <div
      className="css-1vy4sx6"
    />
    <div
      className="css-p5zwuv"
    />
  </div>
</div>
`;
